File: src/Omnia.ProcessManagement.Web/client/fx/models/data/processtypes/ReviewReminder.ts
import { ReviewReminderSchedule, ReviewReminderScheduleTypes, ReviewReminderScheduleFactory } from './reviewreminderschedule';
import { GuidValue, TimePeriodSettings, TimePeriodTypes  } from '@omnia/fx-models';
import { ReviewReminderTask } from '.';

export interface ReviewReminder {
    schedule: ReviewReminderSchedule;
    reminderInAdvance: TimePeriodSettings;

    /**
     * null means no task creation
     * */
    task?: ReviewReminderTask;

    /**
     * who should receive the reminder email
     * There is a spacial value for defining approver : ApproverId - 3d2cbf30-181a-4bf8-afc9-b7a9744bf760
     * */
    personEnterprisePropertyDefinitionIds: Array<GuidValue>;
}

/**
 * NOTE: whenever an new property is added in settings, we need to define its initial value in factory
 * To ensure it fully react on view
 * */
export const ReviewReminderFactory = {
    createDefault(type: ReviewReminderScheduleTypes = ReviewReminderScheduleTypes.TimeAfterPublishing): ReviewReminder {
        let settings: ReviewReminder = {
            schedule: ReviewReminderScheduleFactory.createDefault(type),
            reminderInAdvance: {
                type: TimePeriodTypes.Days,
                value: 30
            },
            personEnterprisePropertyDefinitionIds: [],
            task: null
        }

        return settings;
    }
}

File: src/Omnia.ProcessManagement.Models/ProcessTypes/ReviewReminder.cs
using Omnia.Fx.Models.TimePeriodSettings;
using System;
using System.Collections.Generic;
using System.Text;

namespace Omnia.ProcessManagement.Models.ProcessTypes
{
    public class ReviewReminder
    {
        public ReviewReminderSchedule.ReviewReminderSchedule Schedule { get; set; }
        public TimePeriodSettings ReminderInAdvance { get; set; }

        /// <summary>
        /// null means no task creation
        /// </summary>
        public ReviewReminderTaskSettings Task { get; set; }

        /// <summary>
        /// There is special value for defining approver
        /// <see cref="ProcessTypeItemSettings.ApproverGroupId"/>
        /// </summary>
        public List<Guid> PersonEnterprisePropertyDefinitionIds { get; set; }
    }

    public class ReviewReminderTaskSettings
    {
        public Guid PersonEnterprisePropertyDefinitionId { get; set; }
        public TimePeriodSettings Expiration { get; set; }
    }
}
